src/main/java/appeng/hotkeys/InventoryHotkeyAction.java

```diff
@@ -0,0 +1,44 @@
+package appeng.hotkeys;
+
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+
+import appeng.api.hotkeys.HotkeyAction;
+
+public class InventoryHotkeyAction implements HotkeyAction {
+
+    private final Locatable locatable;
+    private final Opener opener;
+
+    public InventoryHotkeyAction(Locatable locatable, Opener opener) {
+        this.locatable = locatable;
+        this.opener = opener;
+    }
+
+    public InventoryHotkeyAction(Item item, Opener opener) {
+        this((stack) -> stack.getItem() == item, opener);
+    }
+
+    @Override
+    public boolean run(Player player) {
+        var items = player.getInventory().items;
+        for (int i = 0; i < items.size(); i++) {
+            if (locatable.matchItem(items.get(i))) {
+                opener.open(player, i);
```

Comment:
Opener returns boolean, but it seems ignored here?

Reply:
This is a leftover from a previous iteration

but it could be used to keep checking for more items if the first one fails to open

e.g. when having multiple wireless terminals for different systems at different places it would try opening the second one if the first one isn't in range

Reply:
Hmmm, shouldn't the open call be `if(open(...)) return true;` then?